batch/batch/front_end/front_end.py

```diff
@@ -1177,6 +1184,54 @@ async def ui_get_job(request, userdata):
     return await render_template('batch', request, userdata, 'job.html', page_context)
 
 
+@routes.get('/billing_limits')
+@prom_async_time(REQUEST_TIME_GET_BILLING_LIMITS_UI)
+@web_authenticated_users_only()
+async def ui_get_billing_limits(request, userdata):
+    app = request.app
+    db = app['db']
+
+    if not userdata['is_developer']:
+        user = userdata['username']
+    else:
+        user = None
+
+    billing_limits = await _query_billing_projects(db, user=user)
+
+    page_context = {
+        'billing_limits': billing_limits,
+        'is_developer': userdata['is_developer']
+    }
+    return await render_template('batch', request, userdata, 'billing_limits.html', page_context)
+
+
+@routes.post('/billing_limits/{billing_project}/edit')
+@prom_async_time(REQUEST_TIME_POST_BILLING_LIMITS_EDIT_UI)
+@check_csrf_token
+@web_authenticated_developers_only(redirect=False)
+async def post_billing_limits_edit(request, userdata):  # pylint: disable=unused-argument
+    db = request.app['db']
+    billing_project = request.match_info['billing_project']
+    post = await request.post()
+    limit = post['limit']
+
+    session = await aiohttp_session.get_session(request)
+
+    try:
+        if limit == 'None':
+            limit = None
+        else:
+            limit = float(limit)
+            assert limit >= 0
+
+        await db.execute_update('UPDATE billing_projects SET `limit` = %s WHERE name = %s;', (limit, billing_project))
+        set_message(session, f'Modified limit {limit} for billing project {billing_project}.', 'info')
+    except Exception:
+        set_message(session, f'Invalid value {limit} for limit for billing_project {billing_project}', 'error')
```

Comment:
This will also hide any database error as an invalid value message. I prefer:
```
try:
    ...
except Exception:
    set_message(session, f'Invalid ...')
else:
    await db.execute_update(...)
    set_message(session, f'Modified ...')
```
Unless there's a database error that we expect to get and want to treat that as an invalid value?